google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerTableStatements.java

```diff
@@ -0,0 +1,173 @@
+
+/**
+ * Generates the SQL statements for creating and dropping tables in Spanner.
+ */
+public class SpannerTableStatements {
+
+  private static final String CREATE_TABLE_TEMPLATE = "create table {0} ({1}) PRIMARY KEY ({2})";
+
+  private final SpannerDialect spannerDialect;
+
+  public SpannerTableStatements(SpannerDialect spannerDialect) {
+    this.spannerDialect = spannerDialect;
+  }
+
+  /**
+   * Generates the statements needed to drop a table.
+   */
+  public List<String> dropTable(Table table, Metadata metadata) {
+    ArrayList<String> dropStrings = new ArrayList<>();
+
+    Iterator<Index> iteratorIdx = table.getIndexIterator();
+    while (iteratorIdx.hasNext()) {
+      Index curr = iteratorIdx.next();
+      dropStrings.add("drop index " + curr.getName());
+    }
+
+    dropStrings.add(this.spannerDialect.getDropTableString(table.getQuotedName()));
+    return dropStrings;
+  }
+
+  /**
+   * Generates the statements needed to create a table.
+   */
+  public List<String> createTable(Table table, Metadata metadata) {
+
+    Iterable<Column> keyColumns;
+
+    if (table.hasPrimaryKey()) {
+      // a typical table that corresponds to an entity type
+      keyColumns = getSortedPkColumns(table, metadata);
+    } else if (isElementCollection(table, metadata)) {
+      // a table that is actually an element collection property
+      keyColumns = table::getColumnIterator;
+    } else {
+      // the case corresponding to a sequence-table that will only have 1 row.
+      keyColumns = Collections.emptyList();
+    }
+
+    return getCreateTableStrings(table, metadata, keyColumns);
+  }
+
+  /**
+   * Returns true if a table is generated by a Hibernate element collection.
+   */
+  private boolean isElementCollection(Table table, Metadata metadata) {
+    for (Collection collection : metadata.getCollectionBindings()) {
+      if (collection.getCollectionTable().equals(table)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private List<String> getCreateTableStrings(
+      Table table, Metadata metadata, Iterable<Column> keyColumns) {
+
+    // Get the comma separated string of the primary keys of the table.
+    String primaryKeyColNames = StreamSupport.stream(keyColumns.spliterator(), false)
+        .map(Column::getQuotedName)
+        .collect(Collectors.joining(","));
+
+    // Get the comma separated string of all columns of the table.
+    Iterable<Column> columnIterable = () -> (Iterator<Column>) table.getColumnIterator();
+    String allColumnNames = StreamSupport.stream(columnIterable.spliterator(), false)
+        .map(column -> buildColumnTypeString(column, metadata))
+        .collect(Collectors.joining(","));
+
+    ArrayList<String> statements = new ArrayList<>();
+
+    // Build the Create Table string.
+    String createTableString = MessageFormat.format(
+        CREATE_TABLE_TEMPLATE, table.getQuotedName(), allColumnNames, primaryKeyColNames)
+        + getInterleavedClause(table, metadata);
```

Comment:
Consider making interleaved clause part of the formatted message (it can be empty string when missing)